x-pack/legacy/plugins/reporting/export_types/common/lib/screenshots/get_screenshots.ts

```diff
@@ -7,51 +7,34 @@
 import { i18n } from '@kbn/i18n';
 import { HeadlessChromiumDriver as HeadlessBrowser } from '../../../../server/browsers';
 import { LevelLogger } from '../../../../server/lib';
-import { Screenshot, ElementsPositionAndAttribute } from './types';
-
-const getAsyncDurationLogger = (logger: LevelLogger) => {
-  return async (description: string, promise: Promise<any>) => {
-    const start = Date.now();
-    const result = await promise;
-    logger.debug(
-      i18n.translate('xpack.reporting.screencapture.asyncTook', {
-        defaultMessage: '{description} took {took}ms',
-        values: {
-          description,
-          took: Date.now() - start,
-        },
-      })
-    );
-    return result;
-  };
-};
+import { ApmTransaction, Screenshot, ElementsPositionAndAttribute } from './types';
 
 export const getScreenshots = async (
   browser: HeadlessBrowser,
   elementsPositionAndAttributes: ElementsPositionAndAttribute[],
-  logger: LevelLogger
+  logger: LevelLogger,
+  txn: ApmTransaction
 ): Promise<Screenshot[]> => {
   logger.info(
     i18n.translate('xpack.reporting.screencapture.takingScreenshots', {
       defaultMessage: `taking screenshots`,
     })
   );
 
-  const asyncDurationLogger = getAsyncDurationLogger(logger);
   const screenshots: Screenshot[] = [];
 
   for (let i = 0; i < elementsPositionAndAttributes.length; i++) {
+    const apmSpan = txn?.startSpan('get_screenshots', 'read');
     const item = elementsPositionAndAttributes[i];
-    const base64EncodedData = await asyncDurationLogger(
```

Comment:
`asyncDurationLogger` was hiding a type error, because we're resolving a Buffer, not a string.

Reply:
More context about this: since we'll have APM to measure the duration of all these modules, we no longer need this helper function. After having removed it, I noticed the surrounding code treated the helper function return value as a Buffer but it is actually a string of base64 content.